cpu/rp2350/clock.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Tom Hert <git@annsann.eu>
+ * SPDX-FileCopyrightText: 2025 HAW Hamburg
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+#include "periph_cpu.h"
+
+void clock_reset(void) {
+  // Reset the clock system
+  reset_component(RESET_PLL_SYS, RESET_PLL_SYS);
+}
+
+/**
+ * @brief Configures the XOSC and then sets CLK_SYS, PLL_SYS and CLK_PERI to it
+ * @warning Make sure to call clock_reset() before this function to reset the
+ * clock system
+ * @note RP2350 Docs Chapter 8, mostly 8.2 for more details
+ */
+void cpu_clock_init(void) {
+  // Enable the XOSC
+  xosc_start();
+
+  /**
+   * Setup the PLL using the XOSC as the reference clock.
+   */
+  // PLL_SYS->CS = REF_DIV; // Set the reference divider
+  PLL_SYS->FBDIV_INT = 125;  // Set the feedback divider
+
+  /**
+   * Set the post-dividers for the PLL output.
+   */
+  PLL_SYS->PRIM = PDIV;
+  // Turn on PLL
+  atomic_clear(&PLL_SYS->PWR,
+               PLL_PWR_PD_BITS | PLL_PWR_VCOPD_BITS | PLL_PWR_POSTDIVPD_BITS);
+
+  // sleep 10ms to allow the PLL to stabilize
+  xosc_sleep(10);
+
+  // // Wait for lock
+  // while (!(PLL_SYS->CS & PLL_CS_LOCK_BITS)) {
+  //     // Wait for the PLL to lock
+  // }
```

Comment:
This code was intentionally left within the file, technically, at least the way I read through the rp2350 docs it appears like this is something we would want do (Chapter 8 at around ~8.2), however, trying to do this always resulted in the cpu softlocking until I reflashed it with a working binary that did not have this line. I am not really sure why and it appears like other implementations also often ignore this.